docs/reference/analysis/synonyms.asciidoc

```diff
@@ -0,0 +1,261 @@
+
+{es} uses synonyms as part of the <<analysis-overview,analysis process>>.
+There are two different <<analysis-tokenfilters,token filters>> that can be used for including synonyms:
+
+* <<analysis-synonym-tokenfilter>>: Use it for single word synonyms ("car", "automobile").
+* <<analysis-synonym-graph-tokenfilter>>: Use it for synonyms that may span multiple words ("hurriedly", "in a hurry").
+
+Synonyms sets created via synonyms APIs can be included into synonym token filters using `synonyms_set` configuration:
+
+[source,JSON]
+----
+  "filter": {
+    "synonyms_filter": {
+      "type": "synonym_graph",
+      "synonyms_set": "my-synonym-set"
+    }
+  }
+----
+
+The synonym file path needs to be included as part of your token filter using `synonyms_path` configuration:
+
+[source,JSON]
+----
+  "filter": {
+    "synonyms_filter": {
+      "type": "synonym_graph",
+      "synonyms_path": "analysis/synonym-set.txt"
+    }
+  }
+----
+
+Synonyms can be directly included in your token filter using the `synonyms` configuration:
+
+[source,JSON]
+----
+  "filter": {
+    "synonyms_filter": {
+      "type": "synonym_graph",
+      "synonyms": ["pc => personal computer", "computer, pc, laptop"]
+    }
+  }
+----
+
+[discrete]
+[[synonyms-analizers-configure]]
+==== Configure analyzers with synonym token filters
+
+To apply synonyms, you will need to include one of the above synonym token filters into an analyzer, like the following:
+
+[source,JSON]
+----
+      "analyzer": {
+        "my_analyzer": {
+          "type": "custom",
+          "tokenizer": "standard",
+          "filter": ["lowercase", "synonyms_filter"]
+        }
+      }
+----
+
+Order is important for your token filters.
+{es} will use the token filters preceding the synonym filter to parse the synonym rules in a synonyms set.
+
+In the above example, text will be lowercased by the `lowercase` filter before being processed by the `synonyms_filter`.
+This means that all the synonyms defined there needs to be in lowercase, or they won't be found by the synonyms filter.
+
+The synonym rules should not contain words that are removed by a filter that appears after in the chain (like a `stop` filter).
+Removing a term from a synonym rule means there will be no matching for it at query time.
+
+Because entries in the synonym map cannot have stacked positions, some token filters may cause issues here.
+Token filters that produce multiple versions of a token may choose which version of the token to emit when parsing synonyms.
+For example, `asciifolding` will only produce the folded version of the token.
+Others, like `multiplexer`, `word_delimiter_graph` or `ngram` will throw an error.
+
+If you need to build analyzers that include both multi-token filters and synonym filters, consider using the <<analysis-multiplexer-tokenfilter,multiplexer>> filter, with the multi-token filters in one branch and the synonym filter in the other.
+
```

Comment:
I am not sure why we need to duplicate this information here, I think instead of this whole seciton, it is better just to provide a reference to the synonym_filter where all this information is provided in the updated form. Something like this:

Please refer to the synonym_filter documentation to see how this filter interacts with other filters.

Reply:
I moved this information from the synonyms token filters and included it here. I thought it would be better to consolidate common information in a single place.